modules/tensor_mechanics/src/kernels/StressDivergenceRZTensors.C

```diff
@@ -6,28 +6,21 @@
 /****************************************************************/
 
 #include "StressDivergenceRZTensors.h"
-#include "Assembly.h"
-#include "ElasticityTensorTools.h"
+#include "MooseMesh.h"
 #include "libmesh/quadrature.h"
 
 template <>
 InputParameters
 validParams<StressDivergenceRZTensors>()
 {
-  InputParameters params = validParams<StressDivergenceTensors>();
+  InputParameters params = validParams<StressDivergence2DTensors>();
   params.addClassDescription(
       "Calculate stress divergence for an axisymmetric problem in cylinderical coordinates.");
-  params.addRequiredParam<unsigned int>(
-      "component",
-      "An integer corresponding to the direction the variable this kernel acts in. (0 "
-      "for x, 1 for y, 2 for z; note in this kernel disp_x refers to the radial "
-      "displacement and disp_y refers to the axial displacement.)");
-  params.set<bool>("use_displaced_mesh") = true;
   return params;
 }
 
 StressDivergenceRZTensors::StressDivergenceRZTensors(const InputParameters & parameters)
-  : StressDivergenceTensors(parameters)
+  : StressDivergence2DTensors(parameters), _first(!_fe_problem.mesh().hasSecondOrderElements())
```

Comment:
This approach will force the full integration scheme from ABAQUS for all linear meshes--do we want to force that approach?  What about bending problems?

Reply:
I guess it assumes no under integration.  Good point.  Is there a way to know the integration order?